src/payload.ts

```diff
@@ -72,7 +73,12 @@ export class Payload extends StacksMessage {
         bufferArray.push(this.contractAddress.serialize());
         bufferArray.push(this.contractName.serialize());
         bufferArray.push(this.functionName.serialize());
-        // TODO: serialize function args
+        const numArgs = Buffer.alloc(4);
+        numArgs.writeUInt32BE(this.functionArgs.length, 0);
+        bufferArray.push(numArgs)
```

Comment:
@yknl is this the correct way to serialize function args length here?

Reply:
In this case it the result is the same. We should consider if we want to be consistent and use `intToHexString` throughout.